### ISSUE TITLE
In Place Variant

### ISSUE THREAD

**Issue by RoyiAvital** [jw3126/MeanFilters.jl#2]
It will be great to have non allocating in place variant.

**Comment by jw3126** [comment]
yeah, it can be done, we would also need to pass some workspace around in addition to the output array.

**Comment by RoyiAvital** [comment]
It makes sense to expose a buffer.
